Source/JavaScriptCore/inspector/agents/InspectorConsoleAgent.cpp

```diff
@@ -112,14 +105,25 @@ bool InspectorConsoleAgent::developerExtrasEnabled() const
     return m_injectedScriptManager.inspectorEnvironment().developerExtrasEnabled();
 }
 
-void InspectorConsoleAgent::reset()
+void InspectorConsoleAgent::mainFrameNavigated()
 {
-    clearMessages();
+    internalClearMessages(Inspector::Protocol::Console::ClearReason::MainFrameNavigation);
 
     m_times.clear();
     m_counts.clear();
 }
 
+void InspectorConsoleAgent::internalClearMessages(Inspector::Protocol::Console::ClearReason reason)
```

Comment:
NIT: we could just call this `clearMessages` and have the different parameters be enough of a distinguisher :)